crates/matrix-sdk-crypto/src/store/mod.rs

```diff
@@ -697,6 +720,50 @@ impl Store {
 
         Ok(self.tracked_users_cache.iter().map(|u| u.clone()).collect())
     }
+
+    /// Check whether there is a global flag to only encrypt messages for
+    /// trusted devices or for everyone.
+    pub async fn get_only_allow_trusted_devices(&self) -> Result<bool> {
+        let value = self.get_value("only_allow_trusted_devices").await?.unwrap_or_default();
+        Ok(value)
+    }
+
+    /// Set global flag whether to encrypt messages for untrusted devices, or
+    /// whether they should be excluded from the conversation.
+    pub async fn set_only_allow_trusted_devices(
+        &self,
+        block_untrusted_devices: bool,
+    ) -> Result<()> {
+        self.set_value("only_allow_trusted_devices", &block_untrusted_devices).await
+    }
+
+    /// Get custom stored value associated with a key
+    pub async fn get_value<T: DeserializeOwned>(&self, key: &str) -> Result<Option<T>> {
+        let Some(value) = self.get_custom_value(key).await? else {
+            return Ok(None);
+        };
+        let deserialized = self.deserialize_value(&value)?;
+        Ok(Some(deserialized))
+    }
+
+    /// Store custom value associated with a key
+    pub async fn set_value(&self, key: &str, value: &impl Serialize) -> Result<()> {
+        let serialized = self.serialize_value(value)?;
+        self.set_custom_value(key, serialized).await?;
+        Ok(())
+    }
+
+    fn serialize_value(&self, value: &impl Serialize) -> Result<Vec<u8>> {
+        let serialized =
+            rmp_serde::to_vec_named(value).map_err(|x| CryptoStoreError::Backend(x.into()))?;
```

Comment:
The mapped error should ideally be `CryptoStoreError::Serialization`, but this is typed to `serde_json` instead of `rmp_serde` and I dont think adding another `Serialization` case is a good approach. Can we make that serialization error more generic?

Reply:
Yeah, error handling in the stores is messy... I have an idea on how it may be fixable, will post a separate PR.